### PR TITLE
Add table azure_recovery_services_backup_job Closes #663

### LINKED ISSUE #663: Add table azure_backup_jobs
**References**
Azure Backup Jobs are a critical part of management an any Azure Cloud:
https://learn.microsoft.com/en-us/azure/backup/backup-azure-manage-vms

List API:
https://github.com/Azure/azure-sdk-for-go/blob/v58.0.0/services/recoveryservices/mgmt/2021-01-01/backup/jobs.go#L39

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select name, properties from azure_backup_job where vault_name = 'test43-delete-me'
+--------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
| name                                 | properties                                                                                                                                                                                                                          >
+--------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
| 4617f498-e84f-4236-9bb1-7dd4f9f3a6f7 | {"ActivityID":"7b55c0e3-38d5-4cc3-b4c8-33215d2ff010","BackupManagementType":"AzureIaasVM","EndTime":"2023-10-06T06:26:51.4907961Z","EntityFriendlyName":"test-backup","JobType":"AzureIaaSVMJob","Operation":"Undelete","StartTime":>
| c4319ff5-9640-4090-84fb-1f70298fc0a5 | {"ActivityID":"7b55c0e3-38d5-4cc3-b4c8-33215d2ff019","BackupManagementType":"AzureIaasVM","EndTime":"2023-10-06T06:27:59.0919188Z","EntityFriendlyName":"test-backup","JobType":"AzureIaaSVMJob","Operation":"DeleteBackupData","Sta>
| 9e81c8ba-df5a-4aa9-8e80-6906f0fecb58 | {"ActivityID":"346be913-6e41-4b61-902a-84993baf10d6","BackupManagementType":"AzureIaasVM","EndTime":"2023-10-04T15:15:45.1782379Z","EntityFriendlyName":"test-backup","JobType":"AzureIaaSVMJob","Operation":"ConfigureBackup","Star>
| 0cde75e9-72d0-403a-9923-12c1baa2fe8f | {"ActivityID":"5c4398d6-45cd-48f1-9593-a3c2e567edae","BackupManagementType":"AzureIaasVM","EndTime":"2023-10-04T18:16:14.6749904Z","EntityFriendlyName":"test-backup","JobType":"AzureIaaSVMJob","Operation":"Backup","StartTime":"2>
| c85988ea-c528-4b91-b683-ae99bdc35d5b | {"ActivityID":"5d3dc4bb-2899-456c-8464-75eb2fe7c017","BackupManagementType":"AzureIaasVM","EndTime":"2023-10-05T10:50:18.7032339Z","EntityFriendlyName":"test-backup","JobType":"AzureIaaSVMJob","Operation":"DisableBackup","StartT>
| 86b76c54-f012-4832-97be-b9c184aa378b | {"ActivityID":"5d3dc4bb-2899-456c-8464-75eb2fe7c025","BackupManagementType":"AzureIaasVM","EndTime":"2023-10-05T10:52:26.74136Z","EntityFriendlyName":"test-backup","JobType":"AzureIaaSVMJob","Operation":"DeleteBackupData","Start>
+--------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
The PR is in draft state cause of we are encountering an error from Azure Go SDK side while making the API, An issue has been raised on Azure Go SDK side regarding this, once it will get resolved we will comeback to this issue and make it ready for review.

SDK issue reference: https://github.com/Azure/azure-sdk-for-go/issues/21690

**Comment by ParthaI** [comment]
As per the [Azure SDK issue comment](https://github.com/Azure/azure-sdk-for-go/issues/21690#issuecomment-1748945260), the API is working fine and it is returning the result as expected. This PR is good to review now. 👍

File: azure/table_azure_recovery_services_backup_job.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	if rgName != "" {
+		if rgName != strings.Split(*vault.ID, "/")[4] {
+			return nil, nil
+		}
+	}
+
+	subscriptionID := session.SubscriptionID
+	clientFactory, err := armrecoveryservicesbackup.NewBackupJobsClient(subscriptionID, cred, nil)
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_backup_job.listAzureBackupJobs", "NewBackupJobsClient", err)
```
Comment (madhushreeray30) [inline azure/table_azure_recovery_services_backup_job.go:158]:
```suggestion
		plugin.Logger(ctx).Error("azure_recovery_services_backup_job.listAzureBackupJobs", "client_error", err)
```

File: azure/table_azure_recovery_services_backup_job.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+		return nil, nil
+	}
+	pager := clientFactory.NewListPager(*vault.Name, strings.Split(*vault.ID, "/")[4], &armrecoveryservicesbackup.BackupJobsClientListOptions{Filter: nil,
+		SkipToken: nil,
+	})
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("azure_backup_job.listAzureBackupJobs", "api_error", err)
```
Comment (madhushreeray30) [inline azure/table_azure_recovery_services_backup_job.go:167]:
```suggestion
			plugin.Logger(ctx).Error("azure_recovery_services_backup_job.listAzureBackupJobs", "api_error", err)
```

**Review by madhushreeray30 — COMMENTED** [review]
@ParthaI Please take a look at the comments, thanks!

**Review by madhushreeray30 — APPROVED** [review]

File: azure/table_azure_recovery_services_backup_job.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+func listRecoveryServicesAzureBackupJobs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	session, err := GetNewSession(ctx, d, "MANAGEMENT")
+	if err != nil {
+		return nil, err
+	}
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_backup_job.listAzureBackupJobs", "NewDefaultAzureCredential", err)
```
Comment (madhushreeray30) [inline azure/table_azure_recovery_services_backup_job.go:132]:
```suggestion
		plugin.Logger(ctx).Error("azure_recovery_services_backup_job. listRecoveryServicesAzureBackupJobs", "NewDefaultAzureCredential", err)
```

File: azure/table_azure_recovery_services_backup_job.go [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+	if rgName != "" {
+		if rgName != strings.Split(*vault.ID, "/")[4] {
+			return nil, nil
+		}
+	}
+
+	subscriptionID := session.SubscriptionID
+	clientFactory, err := armrecoveryservicesbackup.NewBackupJobsClient(subscriptionID, cred, nil)
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_recovery_services_backup_job.listAzureBackupJobs", "client_error", err)
```
Comment (madhushreeray30) [inline azure/table_azure_recovery_services_backup_job.go:158]:
```suggestion
		plugin.Logger(ctx).Error("azure_recovery_services_backup_job. listRecoveryServicesAzureBackupJobs", "client_error", err)
```

File: azure/table_azure_recovery_services_backup_job.go [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+//// TABLE DEFINITION ////
+
+func tableAzureRecoveryServicesBackupJob(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "azure_recovery_services_backup_job",
+		Description: "Azure Recovery Services Backup Job",
+		List: &plugin.ListConfig{
+			ParentHydrate: listRecoveryServicesVaults,
+			Hydrate:       listRecoveryServicesAzureBackupJobs,
```
Comment (madhushreeray30) [inline azure/table_azure_recovery_services_backup_job.go:23]:
```suggestion
			Hydrate:       listRecoveryServicesBackupJobs,
```

File: azure/table_azure_recovery_services_backup_job.go [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+	Properties armrecoveryservicesbackup.JobClassification
+	Tags       map[string]*string
+	ID         *string
+	Name       *string
+	Type       *string
+}
+
+//// LIST FUNCTION ////
+
+func listRecoveryServicesAzureBackupJobs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (madhushreeray30) [inline azure/table_azure_recovery_services_backup_job.go:124]:
```suggestion
func listRecoveryServicesBackupJobs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

**Pull request merged**